Docker compile.sh error: cannot stat /vearch/build/gamma_build/libgamma

Hi everyone. I've successfully got the image `vearch_env:0.3.1` after executing `compile_env.sh`. But at the execution of `compile.sh` i am getting this error:

```
version info: -X 'main.BuildVersion=0.3' -X 'main.CommitID=9807381d6e0c761756966a6c2be3d8c70c2385a1' -X 'main.BuildTime=2020-07-13 04:07.21'
build gamma
CMake Error: The source directory "/env/app/go/src/github.com/vearch/vearch/engine" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
make: *** No rule to make target `gamma'.  Stop.
build vearch
../client/ps_partition_service.go:22:2: cannot find package "github.com/vearch/vearch/engine/idl/fbs-gen/go/gamma_api" in any of:
        /env/app/go/src/github.com/vearch/vearch/vendor/github.com/vearch/vearch/engine/idl/fbs-gen/go/gamma_api (vendor tree)
        /env/app/go/src/github.com/vearch/vearch/engine/idl/fbs-gen/go/gamma_api (from $GOROOT)
        /root/go/src/github.com/vearch/vearch/engine/idl/fbs-gen/go/gamma_api (from $GOPATH)
build deploy tool
cp: cannot stat '/vearch/build/gamma_build/libgamma.*': No such file or directory
```

`build.sh` is works after that. But container won't start:

```
Creating vearch ... error

ERROR: for vearch  Cannot start service vearch: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"/vearch/bin/vearch\": stat /vearch/bin/vearch: no such file or directory": unknown

ERROR: for vearch  Cannot start service vearch: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"/vearch/bin/vearch\": stat /vearch/bin/vearch: no such file or directory": unknown
ERROR: Encountered errors while bringing up the project.
```

I tried to manually download the gamma of specified branch (which is shown at the main page of this repo):

```bash
[denis@denis-mj vearch]$ git clone git@github.com:vearch/gamma.git
[denis@denis-mj vearch]$ mv gamma engine
[denis@denis-mj vearch]$ cd engine/
[denis@denis-mj engine]$ git checkout f0def6f431
[denis@denis-mj engine]$ cd ../cloud
[denis@denis-mj cloud]$ ./compile.sh
```

But getting this error:

```bash
Scanning dependencies of target gamma
...
[ 88%] Building CXX object CMakeFiles/gamma.dir/realtime/realtime_mem_data.cc.o
[ 92%] Building CXX object CMakeFiles/gamma.dir/profile/profile.cc.o
[ 96%] Building CXX object CMakeFiles/gamma.dir/c_api/gamma_api.cc.o
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1121:36: error: 'DirectMap' in namespace 'faiss' does not name a type
 static void write_direct_map(const faiss::DirectMap *dm, faiss::IOWriter *f) {
                                    ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1121:54: error: ISO C++ forbids declaration of 'dm' with no type [-fpermissive]
 static void write_direct_map(const faiss::DirectMap *dm, faiss::IOWriter *f) {
                                                      ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc: In function 'void tig_gamma::write_direct_map(const int*, faiss::IOWriter*)':
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1123:17: error: request for member 'type' in '* dm', which is of non-class type 'const int'
       (char)dm->type;  // for backwards compatibility with bool
                 ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1125:19: error: request for member 'array' in '* dm', which is of non-class type 'const int'
   WRITEVECTOR(dm->array);
                   ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1093:20: note: in definition of macro 'WRITEVECTOR'
     size_t size = (vec).size();        \
                    ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1125:19: error: request for member 'array' in '* dm', which is of non-class type 'const int'
   WRITEVECTOR(dm->array);
                   ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1076:23: note: in definition of macro 'WRITEANDCHECK'
     size_t ret = (*f)(ptr, sizeof(*(ptr)), n);                                \
                       ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1125:3: note: in expansion of macro 'WRITEVECTOR'
   WRITEVECTOR(dm->array);
   ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1125:19: error: request for member 'array' in '* dm', which is of non-class type 'const int'
   WRITEVECTOR(dm->array);
                   ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1076:37: note: in definition of macro 'WRITEANDCHECK'
     size_t ret = (*f)(ptr, sizeof(*(ptr)), n);                                \
                                     ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1125:3: note: in expansion of macro 'WRITEVECTOR'
   WRITEVECTOR(dm->array);
   ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1126:11: error: request for member 'type' in '* dm', which is of non-class type 'const int'
   if (dm->type == faiss::DirectMap::Hashtable) {
           ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1126:26: error: 'faiss::DirectMap' has not been declared
   if (dm->type == faiss::DirectMap::Hashtable) {
                          ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1129:55: error: request for member 'hashtable' in '* dm', which is of non-class type 'const int'
     const std::unordered_map<idx_t, idx_t> &map = dm->hashtable;
                                                       ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc: In function 'void tig_gamma::write_ivf_header(const faiss::IndexIVF*, faiss::IOWriter*)':
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1141:39: error: cannot convert 'const std::vector<long int>*' to 'const int*' for argument '1' to 'void tig_gamma::write_direct_map(const int*, faiss::IOWriter*)'
   write_direct_map(&ivf->direct_map, f);
                                       ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc: At global scope:
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1155:36: error: variable or field 'read_direct_map' declared void
 static void read_direct_map(faiss::DirectMap *dm, faiss::IOReader *f) {
                                    ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1155:29: error: 'DirectMap' is not a member of 'faiss'
 static void read_direct_map(faiss::DirectMap *dm, faiss::IOReader *f) {
                             ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1155:47: error: 'dm' was not declared in this scope
 static void read_direct_map(faiss::DirectMap *dm, faiss::IOReader *f) {
                                               ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1155:67: error: expected primary-expression before '*' token
 static void read_direct_map(faiss::DirectMap *dm, faiss::IOReader *f) {
                                                                   ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1155:68: error: 'f' was not declared in this scope
 static void read_direct_map(faiss::DirectMap *dm, faiss::IOReader *f) {
                                                                    ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1355:1: error: expected '}' at end of input
 }  // namespace tig_gamma
 ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1136:13: warning: 'void tig_gamma::write_ivf_header(const faiss::IndexIVF*, faiss::IOWriter*)' defined but not used [-Wunused-function]
 static void write_ivf_header(const faiss::IndexIVF *ivf, faiss::IOWriter *f) {
             ^
/env/app/go/src/github.com/vearch/vearch/engine/index/gamma_index_ivfpq.cc:1144:13: warning: 'void tig_gamma::read_index_header(faiss::Index*, faiss::IOReader*)' defined but not used [-Wunused-function]
 static void read_index_header(faiss::Index *idx, faiss::IOReader *f) {
             ^
make[3]: *** [CMakeFiles/gamma.dir/index/gamma_index_ivfpq.cc.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/gamma.dir/all] Error 2
make[1]: *** [CMakeFiles/gamma.dir/rule] Error 2
make: *** [gamma] Error 2
build vearch
# github.com/vearch/vearch/ps/engine/gammacb
../ps/engine/gammacb/gamma.go:20:23: fatal error: gamma_api.h: No such file or directory
 #include "gamma_api.h"
                       ^
compilation terminated.
build deploy tool
cp: cannot stat '/vearch/build/gamma_build/libgamma.*': No such file or directory
```

--- a/cloud/compile.sh
+++ b/cloud/compile.sh
@@ -1,3 +1,7 @@
 #!/usr/bin/env bash
 
+# will download gamma in /engine with correct version
+git submodule init
+git submodule update 
+
 docker run -v $(dirname "$PWD"):/vearch vearch/vearch_env:0.3.1 /vearch/cloud/compile/compile.sh

--- a/cloud/env/env.sh
+++ b/cloud/env/env.sh
@@ -24,7 +24,6 @@ cd /env/app/
 if [ ! -d "faiss" ]; then
     git clone https://github.com/facebookresearch/faiss.git
     cd faiss
-    git checkout v1.6.0
 fi
 cd /env/app/faiss
 ./configure --without-cuda --prefix=/env/app/faiss_install

--- a/docs/Quickstart.md
+++ b/docs/Quickstart.md
@@ -48,7 +48,7 @@ curl -XPUT -H "content-type:application/json" -d '{"name": "test"}' http://127.0
 # create a space in test db which name test too.
 curl -XPUT -H "content-type: application/json" -d' { "name": "test",
 "dynamic_schema": "strict", "partition_num": 2, "replica_num": 1, "engine":
-{"name":"gamma","index_size":10000,	"retrieval_type": "IVFPQ", "retrieval_param": {"metric_type": "InnerProduct","ncentroids": -1,"nsubvector": -1}}, "properties": { "url": { "type": "keyword", "index":"true"}, "feature1": { "type": "vector", "dimension":512, "format": "normalization"}}} ' http://127.0.0.1:4101/space/test/_create
+{"name":"gamma","index_size":10000,	"retrieval_type": "IVFPQ", "retrieval_param": {"metric_type": "InnerProduct","ncentroids": -1,"nsubvector": -1}}, "properties": { "url": { "type": "keyword", "index":true}, "feature1": { "type": "vector", "dimension":512, "format": "normalization"}}} ' http://127.0.0.1:4101/space/test/_create
 ```
 
 A successful response looks like this:
@@ -58,7 +58,7 @@ A successful response looks like this:
 {"code":200,"msg":"success","data":{"id":1,"name":"test"}}
 
 # create space
-{"code":200,"msg":"success","data":{"id":2,"name":"test","version":2,"db_id":1,"enabled":true,"partitions":[{"id":3,"space_id":2,"db_id":1,"partition_slot":0,"replicas":[1]},{"id":4,"space_id":2,"db_id":1,"partition_slot":2147483647,"replicas":[1]}],"partition_num":2,"replica_num":1,"properties":{"url": { "type": "keyword", "index":"true"}, "feature1": { "type": "vector","dimension":512, "format": "normalization"}},"engine":{"name":"gamma","index_size":100000,"max_size":100000,"retrieval_param":{"metric_type": "InnerProduct","ncentroids": -1,"nsubvector": -1}}}}
+{"code":200,"msg":"success","data":{"id":1,"name":"test","version":2,"db_id":1,"enabled":true,"partitions":[{"id":1,"space_id":1,"db_id":1,"partition_slot":0,"replicas":[180]},{"id":2,"space_id":1,"db_id":1,"partition_slot":2147483647,"replicas":[180]}],"partition_num":2,"replica_num":1,"properties":{ "url": { "type": "keyword", "index":true}, "feature1": { "type": "vector", "dimension":512, "format": "normalization"}},"engine":{"name":"gamma","index_size":10000,"max_size":100000,"metric_type":"InnerProduct","retrieval_type":"IVFPQ","retrieval_param":{"metric_type": "InnerProduct","ncentroids": -1,"nsubvector": -1}}}}
 ```
 
 

